### PR TITLE
Add a note to diesel::r2d2 specifying it requires the r2d2 feature

### PR DESCRIPTION
**Pull request by vmalloc**

### CODE REVIEW

File: diesel/src/r2d2.rs [hunk 1]
```diff
@@ -1,4 +1,6 @@
-//! Connection pooling via r2d2
+//! Connection pooling via r2d2.
+//! Note that In order to use this module, you need to have Diesel
```
Comment (weiznich) [inline diesel/src/r2d2.rs]: "Note that *in* order to use this module you need to enable the `r2d2` feature on diesel." 
sounds better for me.

Comment (vmalloc) [reply diesel/src/r2d2.rs]: Oops, my bad. Fixed

File: diesel/src/r2d2.rs [hunk 2]
```diff
@@ -1,4 +1,6 @@
-//! Connection pooling via r2d2
+//! Connection pooling via r2d2.
+//! Note that in order to use this module, you need to have Diesel
```
Comment (sgrif) [inline diesel/src/r2d2.rs]: There should be an empty line here.

File: diesel/src/r2d2.rs [hunk 3]
```diff
@@ -1,4 +1,6 @@
...
+//! installed with the `r2d2` feature
```
Comment (sgrif) [inline diesel/src/r2d2.rs]: I don't like the use of the word "installed" for this. What do you think of "Note: This module requires enabling the `r2d2` feature"

Comment (vmalloc) [reply diesel/src/r2d2.rs]: You're right. That's a lot better.

**Pull request merged**